scripts/wsrep_sst_xtrabackup-v2.sh

```diff
@@ -1814,9 +1817,17 @@ then
         DONOR_MYSQL_VERSION=$(parse_sst_info "$sst_file_info_path" sst mysql-version "")
 
         transition_key=$(parse_sst_info "$sst_file_info_path" sst transition-key "")
+
+        # Use the broken key if the donor version is < 5.7.29 and is not 5.7.28-31.31-41.2
+        # In other words, 5.7.28-31.31-41.2 and above will use the key that was sent
+        if ! check_for_version "$DONOR_MYSQL_VERSION" "5.7.29" &&
```

Comment:
The regex in check_for_version seems to require a version in the `X.Y[.Z[.A]+]` format, while this string is in `X.Y.Z-A.B-U.V`

In other occurences, the donor version was truncated before used with it because of this.

Reply:
The check for version will normalize the string (in X.Y.Z-A.B-U.V to X.Y.Z), in other places the version is embedded with other text so we need to extract the string.  Here, we know the format coming from the donor.

Reply:
The regex in normalize_version doesn't support `-` as a separator, only `.`

Reply:
See comment at https://github.com/percona/percona-xtradb-cluster/blob/5.7/scripts/wsrep_sst_xtrabackup-v2.sh#L1843

Reply:
that is fine.  the version check only does major-minor-patch (5.7.28).  It ignores everything else.  The version check above is looking for anything below 5.7.29